python_modules/dagster/dagster/_core/execution/plan/active.py

```diff
@@ -457,6 +514,8 @@ def handle_event(self, dagster_event: DagsterEvent) -> None:
         step_key = cast(str, dagster_event.step_key)
         if dagster_event.is_step_failure:
             self.mark_failed(step_key)
+            if self._global_concurrency_keys and self._run_id:
+                self._instance.event_log_storage.free_concurrency_slots(self._run_id, step_key)
```

Comment:
does this just eagerly call it for every step instead only for steps with claims ?